doc/src/tutorials/physics/mechanics/duffing-example.rst

```diff
@@ -0,0 +1,146 @@
+- :math:`w`: Width of the Duffing oscillator
+- :math:`d`: Depth of the Duffing oscillator
+- :math:`r`: Radius of the massive bob of the pendulum
+- :math:`q_1`: Generalized coordinate representing the position of the Duffing oscillator
+- :math:`q_2`: Generalized coordinate representing the angle of the pendulum
+- :math:`u_1`: Generalized speed associated with the Duffing oscillator
+- :math:`u_2`: Generalized speed associated with the pendulum
+
+Define Kinematics
+=================
+
+Define all the reference frames and points.
+
+   >>> N = me.ReferenceFrame('N')
+   >>> B = N.orientnew('B', 'axis', (q2, N.z))
+
+The angular velocity of the pendulum in the reference frame is:
+
+   >>> B.ang_vel_in(N)
+   q2'(t) n_z
+
+Locations and velocities of the Duffing Oscillator block and the pendulum are:
+
+   >>> O = me.Point('O')
+   >>> block_point = O.locatenew('block', q1 * N.y)
+   >>> pendulum_point = block_point.locatenew('pendulum', l * B.y)
+
+O is a fixed point in the inertial reference frame.
+
+   >>> O.set_vel(N, 0)
+   >>> block_point.set_vel(N, q1d * N.y)
+   >>> pendulum_point.v2pt_theory(block_point, N, B)
+   q1'(t) n_y + -l*q2'(t) b_x
+
+Define inertia and rigid bodies.
+Here, we assume a simple pendulum which consists of a bob of mass m hanging from a massless string of length l
+and fixed at a pivot point (Duffing Oscillator Block).
+
+   >>> I_block = M / 12 * me.inertia(N, h**2 + d**2, w**2 + d**2, w**2 + h**2)
+   >>> I_pendulum = 2*m*r**2/5*me.inertia(B, 1, 1, 1)
+
+   >>> block_body = me.RigidBody('block', block_point, N, M, (I_block, block_point))
+   >>> pendulum_body = me.RigidBody('pendulum', pendulum_point, B, m, (I_pendulum, pendulum_point))
+
+Define Forces
+=============
+
+We calculate the forces acting on the system.
+
+   >>> path = me.LinearPathway(O, block_point)
+   >>> spring = me.DuffingSpring(k1, k2, path, 0)
+   >>> damper = me.LinearDamper(c1, path)
+
+   >>> loads = spring.to_loads() + damper.to_loads()
+
+   >>> bodies = [block_body, pendulum_body]
+
+   >>> for body in bodies:
+   ...     loads.append(me.Force(body, body.mass * g * N.y))
+
+   >>> loads
+               /      _____           3/2\                  /        _____           3/2\
+         |     /   2       /  2\   |                  |       /   2       /  2\   |
+         \k1*\/  q1   + k2*\q1 /   /*q1               \- k1*\/  q1   - k2*\q1 /   /*q1
+    [(O, ------------------------------ n_y), (block, -------------------------------- n_y), (O, c1*q1'(t) n_y), (block, -c1*q1'(t) n_y), (block, M*g n_y), (pendulum, g*m n_y)]
+                       _____                                         _____
+                      /   2                                         /   2
+                    \/  q1                                        \/  q1
+
+Lagrange's Method
+=================
+
+With the problem setup, the Lagrangian can be calculated, and the equations of motion formed.
+
+   >>> L = me.Lagrangian(N, block_body, pendulum_body)
```

Comment:
The Lagrangian should technically include potential energy term. For this model, it is gravity and the Duffing spring. You can add them as loads, but it is probably worth stating that you are setting the potential energy to zero in the Lagrangian and including the conservative forces in the loads, which is equivalent.